app/services/bulk_import_service.rb

```diff
@@ -182,4 +184,32 @@ def import_lists!
       [row.id]
     end
   end
+
+  def import_account_notes!
+    rows_by_acct = extract_rows_by_acct
+
+    if @import.overwrite?
+      @account.account_notes.reorder(nil).find_each do |account_note|
+        row = rows_by_acct.delete(account_note.target_account.acct)
+
+        if row.nil?
+          account_note.destroy!
+        else
+          row.destroy
+          @import.processed_items += 1
+          @import.imported_items += 1
+          @note = AccountNote.find_or_initialize_by(account: @account, target_account: row.data['acct'])
+          @note.comment = row.data['comment']
+          @note.save! if @note.changed?
+        end
+      end
+
+      # Save pending infos due to `overwrite?` handling
+      @import.save!
+    end
```

Comment:
@ClearlyClaire I think you know about bulk imports, I'm not entirely sure what the logic should be here. I don't really understand what this code is doing to be honest.